src/steps/utils.js

```diff
@@ -185,25 +185,29 @@ export function rewriteUrl(state, url) {
   if (!url || !url.startsWith('https://')) {
     return url;
   }
-  const { pathname, search, hash } = new URL(url);
-
-  if (AZURE_BLOB_REGEXP.test(url)) {
-    const filename = pathname.split('/').pop();
-    const [name, props] = hash.split('?');
-    const extension = name.split('.').pop() || 'jpg';
-    const newHash = props ? `#${props}` : '';
-    return `./media_${filename}.${extension}${newHash}`;
-  }
+  try {
+    const { pathname, search, hash } = new URL(url);
+
+    if (AZURE_BLOB_REGEXP.test(url)) {
+      const filename = pathname.split('/').pop();
+      const [name, props] = hash.split('?');
+      const extension = name.split('.').pop() || 'jpg';
+      const newHash = props ? `#${props}` : '';
+      return `./media_${filename}.${extension}${newHash}`;
```

Comment:
so, it either returns a `path` or a URL? funny, though it was already like this before your change

Reply:
yes. the change here is only that it catches the error when `url` is not valid.